src/compiler/hxb/hxbReader.ml

```diff
@@ -1827,11 +1828,11 @@ class hxb_reader
 				error ("Unexpected type where typedef was expected: " ^ (s_type_path (pack,tname)))
 		))
 
-	method read_mdr =
+	method read_imports =
 		let length = read_uleb128 ch in
 		for _ = 0 to length - 1 do
 			let path = self#read_path in
-			ignore(api#resolve_module path)
+			if read_all then ignore(api#resolve_module path)
```

Comment:
Isn't it pointless to read this chunk at all if `read_all = false`?

Reply:
I wasn't sure if we could skip chunks or not now

Reply:
That has always been possible because all chunks come with their payload size.